monai/handlers/metrics_saver.py

```diff
@@ -0,0 +1,171 @@
+            "*" - generate summary report for every metric_details with all the supported operations.
+            list of strings - generate summary report for every metric_details with specified operations, they
+            should be within this list: [`mean`, `median`, `max`, `min`, `90percent`, `std`].
+            default to None.
+        save_rank: only the handler on specified rank will save to files in multi-gpus validation, default to 0.
+        delimiter: the delimiter charactor in CSV file, default to "\t".
+        output_type: expected output file type, supported types: ["csv"], default to "csv".
+
+    """
+
+    def __init__(
+        self,
+        save_dir: str,
+        metrics: Optional[Union[str, Sequence[str]]] = "*",
+        metric_details: Optional[Union[str, Sequence[str]]] = None,
+        batch_transform: Callable = lambda x: x,
+        summary_ops: Optional[Union[str, Sequence[str]]] = None,
+        save_rank: int = 0,
+        delimiter: str = "\t",
+        output_type: str = "csv",
+    ) -> None:
+        self.save_dir = save_dir
+        self.metrics = ensure_tuple(metrics) if metrics is not None else None
+        self.metric_details = ensure_tuple(metric_details) if metric_details is not None else None
+        self.batch_transform = batch_transform
+        self.summary_ops = ensure_tuple(summary_ops) if summary_ops is not None else None
+        self.save_rank = save_rank
+        self.deli = delimiter
+        self.output_type = output_type
+        self._filenames: List[str] = []
+
+    def attach(self, engine: Engine) -> None:
+        """
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+        engine.add_event_handler(Events.STARTED, self._started)
+        engine.add_event_handler(Events.ITERATION_COMPLETED, self._get_filenames)
+        engine.add_event_handler(Events.EPOCH_COMPLETED, self)
+
+    def _started(self, engine: Engine) -> None:
+        self._filenames = []
+
+    def _get_filenames(self, engine: Engine) -> None:
+        if self.metric_details is not None:
+            _filenames = list(ensure_tuple(self.batch_transform(engine.state.batch)["filename_or_obj"]))
+            self._filenames += _filenames
+
+    def __call__(self, engine: Engine) -> None:
+        """
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+        if not os.path.exists(self.save_dir):
+            os.makedirs(self.save_dir)
+
+        ws = idist.get_world_size()
+        if self.save_rank >= ws:
+            raise ValueError("target rank is greater than the distributed group size.s")
+
+        if self.metrics is not None and len(engine.state.metrics) > 0:
+            if idist.get_rank() == self.save_rank:
+                # only save metrics to file in specified rank
+                with open(os.path.join(self.save_dir, "metrics.csv"), "w") as f:
```

Comment:
would be great to decouple the logic (line 118 - 153) from this specific event handler. Ideally the generic utility should do the conversion:  "metrics data dictionary" => "pandas dataframe" => "to_csv/to_html etc". But I understand we should introduce pandas in a separate pull request properly, so perhaps we could have metrics data dictionary => file (options of csv/html) for now.

Reply:
Sure, I updated the PR to extract a standalone API for the writing logic.
I think it's easy to extend based on Pandas later.
Could you please help review it again? 

Thanks.